di-ipv-core-stub/src/main/java/uk/gov/di/ipv/stub/core/config/uatuser/IdentityMapper.java

```diff
@@ -123,23 +156,72 @@ public Identity mapFormToIdentity(Identity identityOnRecord, QueryParamsMap user
+                        vf,
+                        vu);
+        addresses.add(ukAddress);
+
+        LocalDate secondaryVf =
+                getLocalDate(
+                        userData,
+                        "SecondaryUKAddress.validFrom-year",
+                        "SecondaryUKAddress.validFrom-month",
+                        "SecondaryUKAddress.validFrom-day");
+
+        LocalDate secondaryVu =
+                getLocalDate(
+                        userData,
+                        "SecondaryUKAddress.validUntil-year",
+                        "SecondaryUKAddress.validUntil-month",
+                        "SecondaryUKAddress.validUntil-day");
+
+        UKAddress secondaryUkAddress =
+                new UKAddress(
+                        userData.value("SecondaryUKAddress.buildingNumber"),
+                        userData.value("SecondaryUKAddress.buildingName"),
+                        userData.value("SecondaryUKAddress.street"),
+                        null,
+                        userData.value("SecondaryUKAddress.townCity"),
+                        userData.value("SecondaryUKAddress.postCode"),
+                        secondaryVf,
+                        secondaryVu);
+        if (!Stream.of(
+                        secondaryUkAddress.street(),
+                        secondaryUkAddress.buildingName(),
+                        secondaryUkAddress.buildingNumber(),
+                        secondaryUkAddress.postCode(),
+                        secondaryUkAddress.townCity(),
+                        secondaryUkAddress.county(),
+                        secondaryUkAddress.validFrom(),
+                        secondaryUkAddress.validFrom(),
+                        secondaryUkAddress.validUntil())
+                .allMatch(Objects::isNull)) {
+            addresses.add(secondaryUkAddress);
+        }
 
-        LocalDate dob = LocalDate.of(year, month, dayOfMonth);
+        LocalDate dob =
+                getLocalDate(userData, "dateOfBirth-year", "dateOfBirth-month", "dateOfBirth-day");
         Instant instant = dob.atStartOfDay(ZoneId.systemDefault()).toInstant();
         FindDateOfBirth findDateOfBirth = new FindDateOfBirth(instant, instant);
         FullName fullName = new FullName(userData.value("firstName"), userData.value("surname"));
         return new Identity(
                 identityOnRecord.rowNumber(),
                 identityOnRecord.accountNumber(),
                 identityOnRecord.ctdbDatabase(),
-                ukAddress,
+                addresses,
                 findDateOfBirth,
                 fullName,
                 identityOnRecord.questions());
     }
+
+    private LocalDate getLocalDate(QueryParamsMap userData, String s, String s2, String s3) {
+        if (!Stream.of(userData.value(s), userData.value(s2), userData.value(s3))
+                .allMatch(StringUtils::isBlank)) {
+            int secondaryVfYear = Integer.parseInt(userData.value(s));
```

Comment:
These 3 variables can be removed and the `Integer.parseInt(formData.value("year")` can be inlined when invoking `LocalDate.of()`